### PR TITLE
sleep-config: make hybrid sleep always use 'suspend' disk mode

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: man/systemd-sleep.conf.xml [hunk 1]
```diff
@@ -145,12 +145,9 @@
...
       <varlistentry>
         <term><varname>HibernateMode=</varname></term>
-        <term><varname>HybridSleepMode=</varname></term>
 
-        <listitem><para>The string to be written to <filename>/sys/power/disk</filename> by, respectively,
-        <citerefentry><refentrytitle>systemd-hibernate.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-        and
-        <citerefentry><refentrytitle>systemd-hybrid-sleep.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>.
+        <listitem><para>The string to be written to <filename>/sys/power/disk</filename> by <citerefentry>
+        <refentrytitle>systemd-hibernate.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>.
```
Comment (keszybz) [inline man/systemd-sleep.conf.xml:150]: Hmm, but `systemd-hibernate.service` now also reads this. So maybe do not remove line 153?

File: man/systemd-sleep.conf.xml [hunk 2]
```diff
@@ -160,8 +157,11 @@
         url="https://www.kernel.org/doc/html/latest/admin-guide/pm/sleep-states.html#basic-sysfs-interfaces-for-system-suspend-and-hibernation">the
         kernel documentation</ulink> for more details.</para>
 
-        <para>Note that hybrid sleep corresponds to the <literal>suspend</literal> disk mode. If <varname>HybridSleepMode=</varname>
-        is overridden, you might get plain hibernation instead.</para>
+        <para>
+        <citerefentry><refentrytitle>systemd-hybrid-sleep.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+        corresponds to the <literal>suspend</literal> disk mode. If that's not supported, it will fall back to
```
Comment (keszybz) [inline man/systemd-sleep.conf.xml]: I don't think a service can "correspond" to a suspend mode… "Implies", "uses", something else…?

File: src/shared/sleep-config.c [hunk 3]
```diff
@@ -142,6 +138,9 @@ int parse_sleep_config(SleepConfig **ret) {
                 }
         }
 
+        if (strv_extend_strv(&sc->modes[SLEEP_HYBRID_SLEEP], sc->modes[SLEEP_HIBERNATE], /* filter_duplicates = */ true) < 0)
+                return log_oom();
```
Comment (keszybz) [inline src/shared/sleep-config.c]: Hmm, if `sc->modes[SLEEP_HYBRID_SLEEP]` is now always the same as ` sc->modes[SLEEP_HIBERNATE]`, then maybe just drop it? What is the point of keeping a separate setting?

**Review by keszybz — CHANGES_REQUESTED** [review]

File: src/shared/sleep-config.c (hunk 3 above)
Comment (YHNdnzj) [reply src/shared/sleep-config.c]: No? It is initialized to a strv with only `suspend` (check `sleep_default_mode_table`).

File: man/systemd-sleep.conf.xml (hunk 1 above)
Comment (YHNdnzj) [reply man/systemd-sleep.conf.xml:150]: This is also used by `suspend-then-hibernate` (see below), but we don't explicitly list that here either. I think it's clearer to document this in another paragraph.

**Comment by YHNdnzj** [comment]
Actually, after rethinking about this, there's no point in the "fallback" logic. The user can always choose hibernation as their fallback, and we should make it clear if hybrid-sleep is not actually supported. Reworked this, PTAL.

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
OK. I'll merge this without waiting for the slow CI, because the CI doesn't test hibernation anyway.

**Pull request merged**